doc/Type/List.pod6

```diff
@@ -17,6 +17,33 @@ Arrays to have every value of the list stored in a container.
 C<List> implements C<Positional> and as such provides support for
 L<subscripts|/language/subscripts>.
 
+=head1 Immutability
+
+Lists are immutable objects, i.e., neither the number of elements in a list
+nor the elements themselves can be changed. Thus, it is not possible to use
+operations that change the list structure itself such as L<shift|/routine/shift>,
+L<unshift|/routine/unshift>, L<push|/routine/push>, L<pop|/routine/pop>,
+L<splice|/routine/splice>
+and L<binding|/language/operators#index-entry-Binding_operator>.
+
+    (1, 2, 3).shift;      # Error Cannot call 'shift' on an immutable 'List'
+    (1, 2, 3).unshift(0); # Error Cannot call 'unshift' on an immutable 'List'
+    (1, 2, 3).push(4);    # Error Cannot call 'push' on an immutable 'List'
+    (1, 2, 3).pop;        # Error Cannot call 'pop' on an immutable 'List'
+    (1, 2, 3)[0]:delete;  # Error Cannot remove elements from a List
+    (1, 2, 3)[0] := 0;    # Error Cannot use bind operator with this left-hand side
+    (1, 2, 3)[0] = 0;     # Error Cannot modify an immutable Int
+
+However, this immutability is not deep and thus any of its elements wrapped
```

Comment:
Maybe rephrase? What would deep or shallow immutability be? And also, the list elements themselves are immutable. The content of those elements may vary.

Reply:
I'm taking the definition from [core.c/List.pm6](https://github.com/rakudo/rakudo/blob/add2ec0d491218e59e6c7a517e37458e1a6c3daf/src/core.c/List.pm6#L1-L4). From my understanding, the immutability of list not being deep refers to the fact that this immutability doesn't extend to the elements of structures inside a list that containerize their elements (e.g., Arrays) or to single Scalar containers for that matter. 

Regarding the sentence "Except for the containerized elements, everything else will remain immutable though." what I intended to mean, as you state, was the content of those list elements may vary (if containerized), not necessarily that they can be changed. I was relying on the context provided by the preceding sentence but in hindsight, it's better to remove it since it might cause some confusion.

Reply:
I've rephrased the paragraph, @JJ. Hopefully it's a little better now.